util/oidc/oidc.go

```diff
@@ -367,12 +392,45 @@ func (a *ClientApp) HandleCallback(w http.ResponseWriter, r *http.Request) {
 	if a.secureCookie {
 		flags = append(flags, "Secure")
 	}
-	var claims jwt.MapClaims
+	claims := Claims{}
 	err = idToken.Claims(&claims)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+	groups := claims.Groups
+
+	// Some SSO implementations (Okta) require a call to
+	// the OIDC user info path to get attributes like groups
+	if a.userInfoPath != "" {
+		groups, err = a.GetUserInfoGroups(token.AccessToken, a.settings.IssuerURL(), a.userInfoPath)
+		if err != nil {
+			w.WriteHeader(401)
+			_, _ = w.Write([]byte(fmt.Sprintf("failed to get groups claim: %v", err)))
+			return
+		}
+	}
+
+	// reassamble the JWT claims
+	argoClaims := Claims{
+		groups,
+		jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(a.settings.UserSessionDuration)),
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
+			NotBefore: jwt.NewNumericDate(time.Now()),
+			Issuer:    idToken.Issuer,
```

Comment:
Isn't Argo CD now the issuer? Have you tested that this actually passes verification? Seems like it would attempt to verify against Okta and then fail.

Reply:
As discussed on Slack, I set the issuer to "argocd"